### PR TITLE
feat: Modal closeIcon

### PR DESCRIPTION
**Pull request by afc163**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/18307

### 💡 Background and solution

```diff
<Modal
+ closeIcon={<Icon />}
/>
```

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Modal support custom `closeIcon`. |
| 🇨🇳 Chinese | Modal 新增 `closeIcon` 属性用于自定义关闭图标。 |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/anchor/demo/customizeHighlight.md](https://github.com/ant-design/ant-design/blob/feat-close-icon/components/anchor/demo/customizeHighlight.md)
[View rendered components/anchor/index.en-US.md](https://github.com/ant-design/ant-design/blob/feat-close-icon/components/anchor/index.en-US.md)
[View rendered components/anchor/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/feat-close-icon/components/anchor/index.zh-CN.md)
[View rendered components/descriptions/demo/size.md](https://github.com/ant-design/ant-design/blob/feat-close-icon/components/descriptions/demo/size.md)
[View rendered components/descriptions/index.en-US.md](https://github.com/ant-design/ant-design/blob/feat-close-icon/components/descriptions/index.en-US.md)
[View rendered components/descriptions/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/feat-close-icon/components/descriptions/index.zh-CN.md)
[View rendered components/modal/index.en-US.md](https://github.com/ant-design/ant-design/blob/feat-close-icon/components/modal/index.en-US.md)
[View rendered components/modal/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/feat-close-icon/components/modal/index.zh-CN.md)
[View rendered components/steps/demo/nav.md](https://github.com/ant-design/ant-design/blob/feat-close-icon/components/steps/demo/nav.md)
[View rendered components/steps/demo/simple.md](https://github.com/ant-design/ant-design/blob/feat-close-icon/components/steps/demo/simple.md)
[View rendered components/steps/index.en-US.md](https://github.com/ant-design/ant-design/blob/feat-close-icon/components/steps/index.en-US.md)
[View rendered components/steps/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/feat-close-icon/components/steps/index.zh-CN.md)

### CODE REVIEW

**Comment by chenshuai2144** [comment]
没有 demo 吗？

**Comment by afc163** [comment]
没有

**Review by chenshuai2144 — APPROVED** [review]

File: components/modal/Modal.tsx [hunk 1]
```diff
@@ -220,7 +223,7 @@ export default class Modal extends React.Component<ModalProps, {}> {
       </LocaleReceiver>
     );
 
-    const closeIcon = (
+    const closeIconToRender = closeIcon || (
       <span className={`${prefixCls}-close-x`}>
```
Comment (yesmeck) [inline components/modal/Modal.tsx]: icon 不是应该放在 `span` 里吗？

Comment (afc163) [reply components/modal/Modal.tsx]: 我再起个 PR

Comment (afc163) [reply components/modal/Modal.tsx]: @chenshuai2144 果然还是要有例子才行。

**Pull request merged**